### PR TITLE
CSP-54-web-TrainingVenueDetails

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.Roatp.CourseManagement.Web/Models/ProviderLocations/ProviderLocationViewModel.cs [hunk 1]
```diff
@@ -9,9 +9,15 @@ public class ProviderLocationViewModel
         public int? RegionId { get; set; }
         public string LocationName { get; set; }
         public string VenueNameUrl { get; set; }
+        public string AddressLine1 { get; set; }
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Models/ProviderLocations/ProviderLocationViewModel.cs:12]: This view model was specific to the list. I think we  should leave this as is and have a details view model specifically.

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ViewProviderLocationDetailsControllerTests/ViewProviderLocationDetailsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        private Mock<IUrlHelper> urlHelper;
+        string verifyUrl = "http://test";
+
+        [SetUp]
+        public void Before_Each_Test()
+        {
+            _loggerMock = new Mock<ILogger<ViewProviderLocationDetailsController>>();
+            _mediatorMock = new Mock<IMediator>();
+
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[] { new Claim(ProviderClaims.ProviderUkprn, Ukprn), }, "mock"));
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ViewProviderLocationDetailsControllerTests/ViewProviderLocationDetailsControllerTests.cs:38]: Use test extension method to build context

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ViewProviderLocationDetailsControllerTests/ViewProviderLocationDetailsControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                ControllerContext = new ControllerContext
+                {
+                    HttpContext = httpContext
+                }
+            };
+
+            urlHelper = new Mock<IUrlHelper>();
+
+            UrlRouteContext verifyRouteValues = null;
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ViewProviderLocationDetailsControllerTests/ViewProviderLocationDetailsControllerTests.cs:50]: Use test extension method to mock url context

File: src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/ProviderLocation.cs [hunk 4]
```diff
@@ -7,8 +7,13 @@ public class ProviderLocation
         public Guid NavigationId { get; set; }
         public int? RegionId { get; set; }
         public string LocationName { get; set; }
+        public string AddressLine1 { get; set; }
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Domain/ApiModels/ProviderLocation.cs:10]: When we get list of locations, does it include all these fields? if not, then we should have separate models.

File: src/SFA.DAS.Roatp.CourseManagement.Application/ProviderLocations/Queries/GetProviderLocationDetails/GetProviderLocationDetailsQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        public async Task<GetProviderLocationDetailsQueryResult> Handle(GetProviderLocationDetailsQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider Location request received for Ukprn number {ukprn} and {id}", request.Ukprn, request.Id);
+            var providerLocation = await _apiClient.Get<Domain.ApiModels.ProviderLocation>($"providers/{request.Ukprn}/locations/{request.Id}");
+            if (providerLocation == null)
+            {
+                var message = $"Provider Location not found for {request.Ukprn} and {request.Id}";
+                _logger.LogError(message);
+                throw new ValidationException(message);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Application/ProviderLocations/Queries/GetProviderLocationDetails/GetProviderLocationDetailsQueryHandler.cs:28]: Why ValidationException?

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**